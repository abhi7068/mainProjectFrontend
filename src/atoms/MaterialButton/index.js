import React from 'react';
import { Button } from '@material-ui/core';
import PropTypes from 'prop-types';

const index = (props) => {
  return (
    <div>
      <Button
        data-testid="button"
        type={props.type}
        disabled={props.disabled}
        variant="contained"
        color="primary"
        onClick={props.click}
      >
        {props.name}
      </Button>
    </div>
  );
};
index.propTypes = {
  disabled: PropTypes.bool,
  name: PropTypes.string,
  click: PropTypes.func,
  type: PropTypes.string,
};
export default index;
