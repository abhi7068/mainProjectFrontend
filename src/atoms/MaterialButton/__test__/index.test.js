import React from 'react';

import MaterialButton from '../index';
import { render } from '@testing-library/react';
import renderer from 'react-test-renderer';

it('it renders', () => {
  render(<MaterialButton />);
});
/* it('button component is rendering', () => {
  const { getByTestId } = render(<MaterialButton />);
  expect(getByTestId('button')).render
});
 */
it('it renders with correct text', () => {
  const { getByTestId } = render(<MaterialButton name="Submit" />);
  expect(getByTestId('button')).toHaveTextContent('Submit');
});

it('it matches snapshot', () => {
  const tree = renderer.create(<MaterialButton name="Submit" />).toJSON();
  expect(tree).toMatchSnapshot();
});
