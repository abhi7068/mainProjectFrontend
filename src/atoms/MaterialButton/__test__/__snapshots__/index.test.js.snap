// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`it matches snapshot 1`] = `
<div>
  <button
    className="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary"
    data-testid="button"
    disabled={false}
    onBlur={[Function]}
    onDragLeave={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    onKeyUp={[Function]}
    onMouseDown={[Function]}
    onMouseLeave={[Function]}
    onMouseUp={[Function]}
    onTouchEnd={[Function]}
    onTouchMove={[Function]}
    onTouchStart={[Function]}
    tabIndex={0}
    type="button"
  >
    <span
      className="MuiButton-label"
    >
      Submit
    </span>
  </button>
</div>
`;
